mesmer/stats/_auto_regression.py

```diff
@@ -595,3 +601,339 @@ def _fit_auto_regression_np(data, lags):
+
+
+def _fit_auto_regression_monthly_np(data_month, data_prev_month):
+    """fit an auto regression of lag one (AR(1)) on monthly data - numpy wrapper
+    We use a linear function to relate the independent previous month's
+    data to the dependent current month's data.
+
+    Parameters
+    ----------
+    data_month : np.array
+        A numpy array of the current month's data.
+    data_prev_month : np.array
+        A numpy array of the previous month's data.
+
+    Returns
+    -------
+    slope : :obj:`np.array`
+        The slope of the AR(1) process.
+    intercept : :obj:`np.array`
+        The intercept of the AR(1) proces.
+    """
+
+    def lin_func(indep_var, slope, intercept):
+        return slope * indep_var + intercept
+
+    slope, intercept = scipy.optimize.curve_fit(
+        lin_func,
+        data_prev_month,  # independent variable
+        data_month,  # dependent variable
+        bounds=([-1, -np.inf], [1, np.inf]),
+    )[0]
+
+    return slope, intercept
+
+
+def predict_auto_regression_monthly(ar_params, time, buffer):
+    """deterministically predict time series of an auto regression process
+    with lag one (AR(1)) using individual parameters for each month.
+    This function is deterministic, i.e. does not produce random noise!
+
+    Parameters
+    ----------
+    ar_params : xr.Dataset
+        Dataset containing the estimated parameters of the AR1 process. Must contain the
+        following DataArray objects:
+
+        - intercept
+        - slope
+
+        both of shape (12, n_gridpoints).
+    time : xr.DataArray
+        The time coordinates that determines the length of the predicted timeseries and
+        that will be the assigned time dimension of the predictions.
+    buffer : int
+        Buffer to initialize the auto-regressive process (ensures that start at 0 does
+        not influence overall result).
+
+    Returns
+    -------
+    result : xr.DataArray
+        Predicted deterministic time series of the specified AR(1). The array has
+        shape n_timesteps x n_gridpoints.
+
+    """
+    _check_dataset_form(ar_params, "ar_params", required_vars=("intercept", "slope"))
+
+    (month_dim, gridcell_dim) = ar_params.intercept.dims
+    (n_months, n_gridpoints) = ar_params.intercept.shape
+
+    _check_dataarray_form(
+        ar_params.intercept,
+        "intercept",
+        ndim=2,
+        required_dims=(month_dim, gridcell_dim),
+    )
+    _check_dataarray_form(
+        ar_params.slope, "slope", ndim=2, required_dims=(month_dim, gridcell_dim)
+    )
+    result = _draw_ar_corr_monthly_xr_internal(
+        intercept=ar_params.intercept,
+        slope=ar_params.slope,
+        covariance=None,
+        time=time,
+        realisation=1,
+        seed=0,
+        buffer=buffer,
+        time_dim="time",
+        realisation_dim="realisation",
+    ).squeeze("realisation", drop=True)
+
+    return result
+
+
+def draw_auto_regression_monthly(
+    ar_params,
+    covariance,
```

Comment:
```suggestion
    covariance,
    *,
```

Should we make the other params keyword-only? (You might have to adapt the tests)

Reply:
Yes, then its the same as for the yearly functions, thanks.